### PR TITLE
Throw SQLException in getTime

### LINKED ISSUE #5315: Wrong exception when calling ResultSet.getTimestamp on TIME value
```
java.lang.IllegalArgumentException: Expected column to be a timestamp type but is time(3)
	at io.prestosql.jdbc.AbstractPrestoResultSet.getTimestamp(AbstractPrestoResultSet.java:362)
	at io.prestosql.jdbc.AbstractPrestoResultSet.getTimestamp(AbstractPrestoResultSet.java:327)
	at io.prestosql.jdbc.PrestoResultSet.getTimestamp(PrestoResultSet.java:44)
```

should be `SQLException`

### PR DESCRIPTION
**Pull request by rafatmunshi**
Fixes #5315

### CODE REVIEW

**Comment by rafatmunshi** [comment]
Hi, I sent the signed CLA around 24 hours ago. How much time it would take to add me on file for this PR to be incorporated?
Thanks

**Comment by martint** [comment]
> Hi, I sent the signed CLA around 24 hours ago. How much time it would take to add me on file for this PR to be incorporated?

I'll take care of that later today.

**Comment by ebyhr** [comment]
@cla-bot check

**Comment by rafatmunshi** [comment]
Hey @martint , can you please merge this PR to close this issue?
Thanks

**Comment by martint** [comment]
@rafatmunshi, the commits are part of a single logical, so can you squash them? Also, there are test failures: https://github.com/prestosql/presto/pull/5834/checks?check_run_id=1357228276. E.g.,

```
 TestJdbcResultSetCompatibilityOldServer>BaseTestJdbcResultSet.testTime:232->BaseTestJdbcResultSet.checkRepresentation:616->BaseTestJdbcResultSet.lambda$testTime$31:240 
Expecting:
  <java.lang.IllegalArgumentException: Expected column to be a timestamp type but is time>
to be an instance of:
  <java.sql.SQLException>
but was:
  <"java.lang.IllegalArgumentException: Expected column to be a timestamp type but is time
	at io.prestosql.jdbc.AbstractPrestoResultSet.getTimestamp(AbstractPrestoResultSet.java:362)
	at io.prestosql.jdbc.AbstractPrestoResultSet.getTimestamp(AbstractPrestoResultSet.java:327)
	at io.prestosql.jdbc.PrestoResultSet.getTimestamp(PrestoResultSet.java:43)
	at io.prestosql.jdbc.BaseTestJdbcResultSet.lambda$testTime$30(BaseTestJdbcResultSet.java:239)
	at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:62)
	at org.assertj.core.api.AssertionsForClassTypes.catchThrowable(AssertionsForClassTypes.java:739)
	at org.assertj.core.api.Assertions.catchThrowable(Assertions.java:1168)
	at org.assertj.core.api.Assertions.assertThatThrownBy(Assertions.java:1040)
	at io.prestosql.jdbc.BaseTestJdbcResultSet.lambda$testTime$31(BaseTestJdbcResultSet.java:239)
	at io.prestosql.jdbc.BaseTestJdbcResultSet.checkRepresentation(BaseTestJdbcResultSet.java:616)
	at io.prestosql.jdbc.BaseTestJdbcResultSet.testTime(BaseTestJdbcResultSet.java:232)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.jav
```

**Comment by rafatmunshi** [comment]
@cla-bot check

File: presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java [hunk 1]
```diff
@@ -231,13 +230,12 @@ public void testTime()
...
                 assertThatThrownBy(() -> rs.getDate(column))
                         .isInstanceOf(SQLException.class)
-                        // TODO (https://github.com/prestosql/presto/issues/5313) wrong message
                         .hasMessage(serverSupportsVariablePrecisionTimeWithTimeZone()
                                 ? "Invalid date from server: 00:39:07+01:00"
                                 : "Invalid date from server: 00:39:07.000 +01:00");
                 assertEquals(rs.getTime(column), someBogusValue); // TODO this should fail, as there no java.sql.Time representation for TIME '00:39:07' in America/Bahia_Banderas
                 assertThatThrownBy(() -> rs.getTimestamp(column))
-                        .isInstanceOf(IllegalArgumentException.class) // TODO (https://github.com/prestosql/presto/issues/5315) SQLException
+                        .isInstanceOf(SQLException.class)
```
Comment (ebyhr) [inline presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java:238]: Remove redundant whitespace. 
```suggestion
                        .isInstanceOf(SQLException.class)
```

```bash
2020-11-26T15:30:37.4873074Z [ERROR] src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java:[302] (regexp) RegexpMultiline: Line has trailing whitespace
2020-11-26T15:30:37.4883585Z [ERROR] src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java:[336] (regexp) RegexpMultiline: Line has trailing whitespace
2020-11-26T15:30:37.4893711Z [ERROR] src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java:[351] (regexp) RegexpMultiline: Line has trailing whitespace
2020-11-26T15:30:37.4903756Z [ERROR] src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java:[431] (regexp) RegexpMultiline: Line has trailing whitespace
2020-11-26T15:30:37.4923878Z [ERROR] src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java:[451] (regexp) RegexpMultiline: Line has trailing whitespace
2020-11-26T15:30:37.4934079Z [ERROR] src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java:[464] (regexp) RegexpMultiline: Line has trailing whitespace
```

**Review by ebyhr — COMMENTED** [review]
Please rebase (not merge) on upstream.

**Comment by mosabua** [comment]
:wave: @rafatmunshi - this PR has become inactive. If you're still interested in working on it, please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Pull request closed without merging**